# Notebook 01 - Data Acquisition

Este notebook apenas contém a documentação da estrutura do dataset e código renomeando as colunas.

In [2]:
import pandas as pd

df = pd.read_csv('../data/raw/train.csv')

df.head()

,customer_id,name,age,gender,owns_car,owns_house,no_of_children,net_yearly_income,no_of_days_employed,occupation_type,total_family_members,migrant_worker,yearly_debt_payments,credit_limit,credit_limit_used(%),credit_score,prev_defaults,default_in_last_6months,credit_card_default
0,CST_115179,ita Bose,46,F,N,Y,0.0,107934.04,612.0,Unknown,1.0,1.0,33070.28,18690.93,73,544.0,2,1,1
1,CST_121920,Alper Jonathan,29,M,N,Y,0.0,109862.62,2771.0,Laborers,2.0,0.0,15329.53,37745.19,52,857.0,0,0,0
2,CST_109330,Umesh Desai,37,M,N,Y,0.0,230153.17,204.0,Laborers,2.0,0.0,48416.60,41598.36,43,650.0,0,0,0
3,CST_128288,Rie,39,F,N,Y,0.0,122325.82,11941.0,Core staff,2.0,0.0,22574.36,32627.76,20,754.0,0,0,0
4,CST_151355,McCool,46,M,Y,Y,0.0,387286.00,1459.0,Core staff,1.0,0.0,38282.95,52950.64,75,927.0,0,0,0


## Largando colunas que não possuem utilidade

Irei dropar colunas que acrescentam pouco ou não acrescentam nada ao projeto.

As colunas que serão dropadas são:
* customer_id
* name
* gender

In [3]:
df = df.drop(columns=['customer_id', 'name', 'gender'])

df.head()

,age,owns_car,owns_house,no_of_children,net_yearly_income,no_of_days_employed,occupation_type,total_family_members,migrant_worker,yearly_debt_payments,credit_limit,credit_limit_used(%),credit_score,prev_defaults,default_in_last_6months,credit_card_default
0,46,N,Y,0.0,107934.04,612.0,Unknown,1.0,1.0,33070.28,18690.93,73,544.0,2,1,1
1,29,N,Y,0.0,109862.62,2771.0,Laborers,2.0,0.0,15329.53,37745.19,52,857.0,0,0,0
2,37,N,Y,0.0,230153.17,204.0,Laborers,2.0,0.0,48416.60,41598.36,43,650.0,0,0,0
3,39,N,Y,0.0,122325.82,11941.0,Core staff,2.0,0.0,22574.36,32627.76,20,754.0,0,0,0
4,46,Y,Y,0.0,387286.00,1459.0,Core staff,1.0,0.0,38282.95,52950.64,75,927.0,0,0,0


## Renomeando as colunas

In [4]:
colunas_nomes = {
    "age": "idade",
    "owns_car": "tem_carro",
    "owns_house": "tem_casa",
    "no_of_children": "numero_de_filhos",
    "net_yearly_income": "renda_anual",
    "no_of_days_employed": "dias_no_emprego",
    "occupation_type": "tipo_de_profissão",
    "total_family_members": "total_membros_familia",
    "migrant_worker": "imigrante",
    "yearly_debt_payments": "pagamento_anual_de_debitos",
    "credit_limit": "limite",
    "credit_limit_used(%)":"porcentagem_uso_credito",
    "prev_defaults":"atrasos_anteriores",
    "default_in_last_6months": "atrasos_nos_ultimos_6meses",
    "credit_card_default": "inadimplencia"

}

df.rename(columns=colunas_nomes, inplace=True)

df.head()

,idade,tem_carro,tem_casa,numero_de_filhos,renda_anual,dias_no_emprego,tipo_de_profissão,total_membros_familia,imigrante,pagamento_anual_de_debitos,limite,porcentagem_uso_credito,credit_score,atrasos_anteriores,atrasos_nos_ultimos_6meses,inadimplencia
0,46,N,Y,0.0,107934.04,612.0,Unknown,1.0,1.0,33070.28,18690.93,73,544.0,2,1,1
1,29,N,Y,0.0,109862.62,2771.0,Laborers,2.0,0.0,15329.53,37745.19,52,857.0,0,0,0
2,37,N,Y,0.0,230153.17,204.0,Laborers,2.0,0.0,48416.60,41598.36,43,650.0,0,0,0
3,39,N,Y,0.0,122325.82,11941.0,Core staff,2.0,0.0,22574.36,32627.76,20,754.0,0,0,0
4,46,Y,Y,0.0,387286.00,1459.0,Core staff,1.0,0.0,38282.95,52950.64,75,927.0,0,0,0


## Verificando integridade e estrutura

In [5]:
df.shape

(45528, 16)

In [6]:
df.columns.tolist()

['idade',
 'tem_carro',
 'tem_casa',
 'numero_de_filhos',
 'renda_anual',
 'dias_no_emprego',
 'tipo_de_profissão',
 'total_membros_familia',
 'imigrante',
 'pagamento_anual_de_debitos',
 'limite',
 'porcentagem_uso_credito',
 'credit_score',
 'atrasos_anteriores',
 'atrasos_nos_ultimos_6meses',
 'inadimplencia']

In [7]:
df.dtypes

idade                           int64
tem_carro                      object
tem_casa                       object
numero_de_filhos              float64
renda_anual                   float64
dias_no_emprego               float64
tipo_de_profissão              object
total_membros_familia         float64
imigrante                     float64
pagamento_anual_de_debitos    float64
limite                        float64
porcentagem_uso_credito         int64
credit_score                  float64
atrasos_anteriores              int64
atrasos_nos_ultimos_6meses      int64
inadimplencia                   int64
dtype: object

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45528 entries, 0 to 45527
Data columns (total 16 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   idade                       45528 non-null  int64  
 1   tem_carro                   44981 non-null  object 
 2   tem_casa                    45528 non-null  object 
 3   numero_de_filhos            44754 non-null  float64
 4   renda_anual                 45528 non-null  float64
 5   dias_no_emprego             45065 non-null  float64
 6   tipo_de_profissão           45528 non-null  object 
 7   total_membros_familia       45445 non-null  float64
 8   imigrante                   45441 non-null  float64
 9   pagamento_anual_de_debitos  45433 non-null  float64
 10  limite                      45528 non-null  float64
 11  porcentagem_uso_credito     45528 non-null  int64  
 12  credit_score                45520 non-null  float64
 13  atrasos_anteriores          455

## Resumo final

* Número de colunas: 16
* Número de linhas: 45528
* Nomes das colunas:
'idade',
 'tem_carro',
 'tem_casa',
 'numero_de_filhos',
 'renda_anual',
 'dias_no_emprego',
 'tipo_de_profissão',
 'total_membros_familia',
 'imigrante',
 'pagamento_anual_de_debitos',
 'limite',
 'porcentagem_uso_credito',
 'credit_score',
 'atrasos_anteriores',
 'atrasos_nos_ultimos_6meses',
 'inadimplencia'
